marshal/filesystem.go

```diff
@@ -29,35 +27,25 @@ import (
 // Filesystem output is a list of filesystem entries with positional attribute columns
 // separated by one or more spaces
 func Filesystems(cmdOutput []byte) []model.Filesystem {
+	filesystems := []model.Filesystem{}
 
-	lines := "["
 	scanner := bufio.NewScanner(strings.NewReader(string(cmdOutput)))
+
 	for scanner.Scan() {
-		lines += "{"
 		line := scanner.Text()
-		line = strings.Join(strings.Fields(line), " ")
-		splitted := strings.Split(line, " ")
-		lines += marshalKey("filesystem") + marshalString(strings.TrimSpace(splitted[0])) + ", "
-		lines += marshalKey("fstype") + marshalString(strings.TrimSpace(splitted[1])) + ", "
-		lines += marshalKey("size") + marshalString(strings.TrimSpace(splitted[2])) + ", "
-		lines += marshalKey("used") + marshalString(strings.TrimSpace(splitted[3])) + ", "
-		lines += marshalKey("available") + marshalString(strings.TrimSpace(splitted[4])) + ", "
-		lines += marshalKey("usedperc") + marshalString(strings.TrimSpace(splitted[5])) + ", "
-		lines += marshalKey("mountedon") + marshalString(strings.TrimSpace(splitted[6])) + ", "
-		lines += "},"
-	}
-
-	lines += "]"
-	lines = strings.Replace(lines, ", }", "}", -1)
-	lines = strings.Replace(lines, "},]", "}]", -1)
-
-	b := []byte(lines)
-	var m []model.Filesystem
-	err := json.Unmarshal(b, &m)
-
-	if err != nil {
-		log.Fatal(err)
+		iter := newIter(strings.Fields(line))
+
+		fs := model.Filesystem{}
+
+		fs.Filesystem = strings.TrimSpace(iter())
+		fs.Type = strings.TrimSpace(iter())
+		fs.Size = TrimParseInt64(iter())
+		fs.UsedSpace = TrimParseInt64(iter())
+		fs.AvailableSpace = TrimParseInt64(iter())
```

Comment:
Are you sure with the iter solution? We have lost some flexibility

Reply:
I like it because:
- things here are in order, you don't have to bother about the index, add useless complexity to the reader and it's prone for errors
- you could do it with an `i++`... But not in Go!
- you can insert and remove things without rewrite every index manually, that sucks
- you still have the flexibility that you want: just throw away the iter and return to use the index, if needed!

Reply:
Ok